### DISCUSSION TITLE
Multiple unnecessary queries calling when we trigger event

### DISCUSSION THREAD

**Discussion by arjundexbytes** [livewire/livewire#8990] [7. Bugs (archived)]
### Livewire version

3.3.5

### Laravel version

10.40

### Which PHP version are you using?

PHP 8.3

### Steps To Reproduce

![image](https://github.com/user-attachments/assets/3ba78410-f9c8-4e35-afce-a662e1c296ef)

I'm working on a Livewire 3 component called CreateOrder, which handles data such as countries, vendors, and other related information. I’ve noticed that the component seems to make multiple unnecessary queries to the countries table when loading a select option in my form, despite the IDs being preloaded.

### Repository or Wirebox URL

https://wirebox.app/b/4llre

### Additional information

multiple redundant queries are being triggered, particularly with the countries table, even though the data should already be loaded. This is causing unnecessary database load and could impact the form’s performance.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by lrljoe** [comment]
Your wirebox does not present your issue at all.

Please share a proper wirebox with all of the code required to replicate the issue.

![image](https://github.com/user-attachments/assets/dd8039c0-5e3f-4502-95ca-a9c6cf0bb8ff)


Keep in mind how Livewire v3 works with Models etc.

Provide a proper working example/wirebox of the issue.

Otherwise nobody can actually offer any help whatsoever.

Looking at what's being called, I suspect you're trying to load a collection of Models into a public property.  Therefore every time a lifecycle occurs, you're going to go back to the database to re-populate it.

**Comment by arjundexbytes** [comment]
I have updated the code in the wirebox to reflect the issue accurately. It now includes all the necessary code required to replicate the problem.

In this updated version, I’ve addressed the potential issue you mentioned regarding collecting models on public property, ensuring that the database calls are minimized. Please let me know if you still notice any gaps or have further suggestions.
